### PR TITLE
WIP: 548 postgresql indexer

### PR DESCRIPTION
**Pull request by melekes**
See #548

### CODE REVIEW

File: state/txindex/postgresql/postgresql.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		return errors.Wrap(err, "failed to begin PG transaction")
+	}
+
+	stmt, err := txn.Prepare(pq.CopyIn("txs", "hash", "height", "index", "tx", "result_data", "result_code", "result_log"))
+	if err != nil {
+		return errors.Wrap(err, "failed to prepare PG statement")
+	}
+
+	for _, result := range b.Ops {
+		_, err = stmt.Exec(fmt.Sprintf("%X", result.Tx.Hash()), int64(result.Height), int64(result.Index), result.Tx, result.Result.Data, int64(result.Result.Code), result.Result.Log)
```
Comment (odeke-em) [inline state/txindex/postgresql/postgresql.go]: Looks unsafe to me, perhaps we should do `"? ? ? ? ?", fmt.Sprintf("%x"....`

**Comment by ebuchman** [comment]
@melekes What should we do with this ? Hard to see the core changes amidst the pubsub stuff

**Comment by ebuchman** [comment]
Oh maybe we just need to remove that first commit ?

**Comment by melekes** [comment]
> What should we do with this ?

this thing probably needs more work

**Comment by melekes** [comment]
- [ ] index Tags
- [ ] config for Postgresql connection?

**Comment by ebuchman** [comment]
Closing in favour of focused issue: https://github.com/tendermint/tendermint/issues/1161. Let's keep the branch around so we can re-open when we need to. 

Don't think we need it for launch, but probably shortly after.

**Pull request closed without merging**